packages/datatrak-web/src/views/AccountSettingsPage/RequestCountryAccessSection/RequestCountryAccessForm.tsx

```diff
@@ -0,0 +1,244 @@
+ */
+
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { useForm } from 'react-hook-form';
+import { Box as MuiBox, FormLabel as MuiFormLabel, useMediaQuery } from '@material-ui/core';
+import { Checkbox, Form, FormInput, TextField } from '@tupaia/ui-components';
+import { Country, TupaiaWebCountryAccessListRequest } from '@tupaia/types';
+import { ensure } from '@tupaia/tsutils';
+import { Button } from '../../../components';
+import { errorToast, successToast } from '../../../utils';
+import { theme } from '../../../theme';
+import { useCountryAccessList, useCurrentUser, useRequestProjectAccess } from '../../../api';
+
+const StyledForm = styled(Form)`
+  inline-size: 100%;
+  ${theme.breakpoints.up('md')} {
+    max-inline-size: 44.25rem;
+  }
+`;
+
+const StyledFieldset = styled.fieldset`
+  border: none;
+  margin: 0;
+  padding: 0;
+
+  display: grid;
+  gap: 1.25rem;
+  grid-auto-flow: column;
+  grid-template: auto auto / auto;
+
+  ${theme.breakpoints.up('sm')} {
+    grid-template: auto / 1fr 1fr;
+  }
+
+  .MuiFormLabel-root {
+    color: ${props => props.theme.palette.text.primary};
+    font-weight: ${theme.typography.fontWeightMedium};
+  }
+
+  // Fix labels appearing over hamburger menu drawer
+  .MuiInputLabel-outlined {
+    z-index: auto;
+  }
+`;
+
+const CountryChecklistWrapper = styled(MuiBox)`
+  display: block flex;
+  flex-direction: column;
+  height: 18.625rem;
+`;
+
+/** Matches styling of .MuiFormLabel-root in ui-components TextField */
+const StyledFormLabel = styled(MuiFormLabel)`
+  font-size: 0.9375rem;
+  line-height: 1.125rem;
+  margin-bottom: 3px;
+`;
+
+const CountryChecklist = styled.fieldset`
+  margin: 0;
+  padding-block: 0;
+
+  border-radius: 0.1875rem;
+  border: 1px solid ${props => props.theme.palette.grey[400]};
+  block-size: 100%;
+  overflow-y: scroll; /* fallback */
+  overflow-block: scroll;
+  padding-inline: 0.87rem;
+
+  :disabled {
+    background-color: #f5f5f5;
+  }
+`;
+
+const StyledCheckbox = styled(Checkbox)`
+  margin-block: 0;
+
+  .MuiFormControlLabel-root {
+    inline-size: 100%;
+  }
+
+  .MuiFormControlLabel-label {
+    font-size: 0.875rem;
+    inline-size: 100%;
+    line-height: 1.125rem;
+  }
+`;
+
+const StyledBox = styled(MuiBox)`
+  display: block flex;
+  flex-direction: column;
+  gap: 1.25rem;
+`;
+
+// Usage of this component has inline styling. See there for explanation.
+const StyledFormInput = styled(FormInput)`
+  flex-grow: 1;
+  margin: 0;
+
+  .MuiInputBase-root {
+    align-items: start;
+    block-size: 100%;
+    max-block-size: 100%;
+  }
+
+  .MuiInputBase-input {
+    box-sizing: border-box;
+  }
+`;
+
+interface RequestCountryAccessFormFields {
+  countryIds: Country['id'][];
+  reasonForAccess?: string;
+}
+
+export const RequestCountryAccessForm = () => {
+  const { project } = useCurrentUser();
+  const projectCode = project?.code;
+
+  const queryResult = useCountryAccessList();
```

Comment:
I'd be a fan of just restructuring the values we use here, instead of using `queryResult` and then `queryResult.data` etc